userspace/libscap/engine/bpf/bpf_public.h

```diff
@@ -16,11 +13,21 @@ limitations under the License.
 
 #pragma once
 
+#include <stdint.h>
+
+#define BPF_ENGINE "bpf"
+#define BPF_ENGINE_LEN 4
+
 #ifdef __cplusplus
-extern "C" {
+extern "C"
+{
 #endif
 
-const char *scap_get_bpf_probe_from_env();
+	struct scap_bpf_engine_params
+	{
+		uint64_t single_buffer_dim; ///<  dim of a single shared buffer. Usually, we have one buffer for every online CPU.
```

Comment:
Left comments in https://github.com/falcosecurity/falco/pull/2164.
Therefore only posting a small cross-reference here: Trusting end user input typically is terrifying especially for such an important parameter. How about brainstorming on what a sanitization logic could look like? Could even include correcting the value to the nearest mathematically correct value? CC @gnosek

Reply:
yeah, absolutely we need to check that these values are correct. As I said in [Falco](https://github.com/falcosecurity/falco/pull/2164#discussion_r949911057) if we use the number of pages we probably need to check only that is a positive value and that it is a power of 2. The sanitization logic is a good idea, not sure if defaulting it to the nearest value or using the default value since the provided one is not correct :thinking: 

This PR just introduces this field in the `scap-open` args, I will allow the driver to use it in another one, so probably i will put the checks in that PR, BTW having a clear idea on how to proceed from the beginning seems reasonable to me!

P.S @gnosek @FedeDP  do you remember if also for the kmod we need a buffer dimension that is a power of 2?

Reply:
Basically the only requirement I'm aware of is that it needs to be mmapped back-to-back twice without gaps, so it must be a multiple of page size (usually 4K, up to 16 or 64? on some arches). Same goes for udig, BTW.

Reply:
Good to know, thanks @gnosek! @Andreagit97 could see how correcting to nearest correct value can indeed terribly backfire. Suggested hard-coding an array of 10-12 values to choose from in the other PR. That way nothing should go wrong in the sanitization check and it will be simple.

Reply:
Uhm I do not completely agree with a fixed array because otherwise every time someone wants to try a new dimension we have to open a new pull request, and we lose in flexibility, maybe upper/lower bound checks are enough :thinking: BTW I will move this conversation on the Falco PR, the implementation/validation of variable size buffers will take place in another PR here we have introduced only the argument and since this PR it's quite blocking I wouldn't stop all the flow for this reason, WDYT?